Dockerfile

```diff
@@ -11,36 +11,32 @@ ARG COMPONENT
 ARG GOPROXY_ARG
 ENV GOPROXY=${GOPROXY_ARG}
 WORKDIR /workspace
-COPY "$COMPONENT"/go.* ./
+COPY "$COMPONENT" "$COMPONENT"
```

Comment:
can we do `COPY "$COMPONENT"/go.* ./"$COMPONENT"/`, that way this base stage gets invalidated only when the module dependencies or the client module code changes

Reply:
If we consider the "base" layer to provide set of prerequisites to subsequent layers, then I think it may be good to copy the entire component directory. If we think of the "base" layer to be strictly used for module dependency resolutions, then every other layer will need to COPY the directory. Thoughts?

Reply:
Added changes to copy component code within a separate layer to avoid mounting entire context and keeping the base layer strictly around dependency resolution.